src/python_testing/TC_IDM_1_4.py

```diff
@@ -0,0 +1,167 @@
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+
+from dataclasses import dataclass
+
+import chip.clusters as Clusters
+from chip import ChipUtility
+from chip.exceptions import ChipStackError
+from chip.interaction_model import InteractionModelError, Status
+from matter_testing_support import MatterBaseTest, async_test_body, default_matter_test_main, type_matches
+from mobly import asserts
+
+
+@dataclass
+class FakeInvalidBasicInformationCommand(Clusters.BasicInformation.Commands.MfgSpecificPing):
+    @ChipUtility.classproperty
+    def must_use_timed_invoke(cls) -> bool:
+        return False
+
+
+class TC_IDM_1_4(MatterBaseTest):
+
+    @async_test_body
+    async def test_TC_IDM_1_4(self):
+        dev_ctrl = self.default_controller
+        dut_node_id = self.dut_node_id
+
+        self.print_step(0, "Commissioning - already done")
+
+        self.print_step(1, "Get remote node's MaxPathsPerInvoke")
+        session_parameters = dev_ctrl.GetRemoteSessionParameters(dut_node_id)
+        max_paths_per_invoke = session_parameters.maxPathsPerInvoke
+
+        asserts.assert_greater_equal(max_paths_per_invoke, 1, "Unexpected error returned from unsupported endpoint")
+        asserts.assert_less_equal(max_paths_per_invoke, 65535, "Unexpected error returned from unsupported endpoint")
+
+        self.print_step(2, "Sending `MaxPathsPerInvoke + 1` InvokeRequest if it fits into single MTU")
+        # In practice, it was noticed that we could only fit 57 commands before we hit the MTU limit as a result we
+        # conservatively try putting up to 100 commands into an Invoke request.
+        cap_for_batch_commands = 100
+        number_of_commands_to_send = min(max_paths_per_invoke + 1, cap_for_batch_commands)
+
+        invalid_command_id = 0xffff_ffff
+        list_of_commands_to_send = []
+        for endpoint_index in range(number_of_commands_to_send):
+            invalid_command = Clusters.BasicInformation.Commands.MfgSpecificPing()
+            invalid_command.command_id = invalid_command_id
+
+            list_of_commands_to_send.append(Clusters.Command.InvokeRequestInfo(endpoint_index, invalid_command))
+
+        asserts.assert_greater_equal(len(list_of_commands_to_send), 2,
+                                     "Step 2 is always expected to try sending at least 2 command, something wrong with test logic")
+        try:
+            await dev_ctrl.SendBatchCommands(dut_node_id, list_of_commands_to_send)
+            # If you get the assert below it is likely because cap_for_batch_commands is actually too low.
+            # This might happen after TCP is enabled and DUT supports TCP.
+            asserts.fail("Unexpected success return from sending too many commands")
```

Comment:
Can you add the max_paths_per_invoke and cap into the error message here? 

Also, once https://github.com/project-chip/matter-test-scripts/pull/52 is in, it might be good to put together a doc outlining what you think some of common failures in this test might be. It seems like it could be complicated to debug, and giving the ATLs some guidance might save you getting pings down the road.

Reply:
Will do